### DISCUSSION TITLE
561M model weights posted

### DISCUSSION THREAD

**Discussion by HarleyCoops** [karpathy/nanochat#37] [General]
Let me save you $250! The mid run (10 hour) is done and I have posted model weights here: https://huggingface.co/HarleyCooper/nanochat

Working on inference now. 

Enjoy!

**Comment by karpathy** [comment]
Nice. I trained a d26 overnight (the ~$300 tier that outperforms GPT-2 CORE) that looks like this:

---

## Summary

- Characters: 336,769
- Lines: 8,216
- Files: 43
- Tokens (approx): 84,192
- Dependencies (uv.lock lines): 2,004

| Metric          | BASE     | MID      | SFT      | RL       |
|-----------------|----------|----------|----------|----------|
| CORE            | 0.2738   | -        | -        | -        |
| ARC-Challenge   | -        | 0.2918   | 0.3140   | -        |
| ARC-Easy        | -        | 0.3864   | 0.4592   | -        |
| GSM8K           | -        | 0.0417   | 0.0758   | -        |
| HumanEval       | -        | 0.0732   | 0.0793   | -        |
| MMLU            | -        | 0.3076   | 0.3299   | -        |
| ChatCORE        | -        | 0.0859   | 0.1252   | -        |

Total wall clock time: 12h5m

---

And my $1000 tier d32 will probably finish late today.

I think given there's a bunch of interest in inference without training, I'll spend some time today figuring out a good model export/import pipeline, and similar to you as well possibly the huggingface compatibility part.

**Reply by Seqaeon** [reply]
Hello Sir,

Can we have like a file that tracks the current performance (bpp,core) at various depths for runs you have done?

**Comment by simonw** [comment]
Since the resulting 561M parameter models are pretty tiny I'm confident there will be a lot of interest in CPU inference as well.

**Comment by burtenshaw** [comment]
hey @HarleyCoops, nice work on the model. 

I've also trained a d20 [base model](https://huggingface.co/nanochat-students/base-d20) and got the inference code for transformers working. Next, I'll move on the mid training with transformers integration.

@karpathy I'm sharing everything on [nanochat-students](https://huggingface.co/nanochat-students). I think it would be cool if we could the pretrain and mid train weights on to the hub/transformers, so that folk can try out the post-training code with less resources.  

Happy to help.

**Comment by HarleyCoops** [comment]
Thank you @burtenshaw ! I barely know what I am doing and am here to learn and help. Such a great gift by @karpathy. I am thinking about an RL version too. I would love to wire in some of DeepMind's mech interp tools to watch how the heads move.  There is so much room for activities!

**Comment by burtenshaw** [comment]
Rad @HarleyCoops ! Feel free to push any models or demos to https://huggingface.co/nanochat-students.

I've copied the inference code into a transformers compatible format for this model: https://huggingface.co/nanochat-students/chat-d20 . I'll now port it to a transformers fork. I also deployed a [demo](https://huggingface.co/spaces/nanochat-students/chat-d20-demo) of the same chat model.

**Comment by HarleyCoops** [comment]
Hey @burtenshaw I just published a new RL version on a different dataset. I didnt get the performance bump i hoped but is it cool if I pull this model into the nanochat student org?  Its an interesting pipeline to adapt nanochat RL to other HF datasets. 

https://huggingface.co/HarleyCooper/nanochatAquaRat

@karpathy Is the idea here that I need to go back to mid training and add layers specific to the new dataset in order to get a real RL boost? It seems nanochat learned everything it needed to know about algebra in SFT.

**Comment by HarleyCoops** [comment]
I should be able to build that, sure. I’ll post it.
